### DISCUSSION TITLE
n-form - Error messages from server

### DISCUSSION THREAD

**Discussion by devstack-be** [tusen-ai/naive-ui#3007] [General]
Hello,
First of all, thank you for this great library.
Is it possible, to enter the possible validation error messages coming from the server?
I don't see any function or anything on the api form to assign error messages.

Thanks a lot.

**Comment by 07akioni** [comment]
Does this demo meet your demand?

https://www.naiveui.com/en-US/os-theme/components/form#async.vue

**Reply by devstack-be** [reply]
Hello, thanks for your answer.
I don't think so. This a validation with a promise for each field, before sending data to server. Here I would like to be able to add error(s) message(s), after the request has been sent to the server. Per example doing something like this:
axios.post('...', data).then((response) => ).catch((error) => refForm.value.setErrors({name: ...}, ...}))

Thanks a lot

**Reply by 07akioni** [reply]
I think you may use custom trigger to do it (although a bit cumbersome)

```
const validatorErrorFromServerRef = ref()

axios.post().then(() => {
  formRef.value.validate(
    undefined,
    rule => rule.trigger === 'async'
  )
})

rules = [
  ...
  {
    trigger: 'async',
    validator: () => {
      return validatorErrorFromServerRef.value[field] || true
    }
  }
  ...
]
```

**Reply by 07akioni** [reply]
<img width="1422" alt="image" src="https://user-images.githubusercontent.com/18677354/170887409-04ac9bdd-bac3-45a0-a10c-ed4528f428f6.png">

**Reply by chioio** [reply]
I has the same problem, I want to trigger error message from server when submit passed the rules, but got some filed error.
